libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/ElapsedTime.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.services.android.navigation.v5.exception.NavigationException
+import kotlin.math.roundToLong
+
+internal class ElapsedTime {
+    var start: Long? = null
+        private set
+    var end: Long? = null
+        private set
+
+    companion object {
+        private const val ELAPSED_TIME_DENOMINATOR = 1e+9
+        private const val PRECISION = 100.0
+    }
+
+    val elapsedTime: Double
+        get() {
+            if (start == null || end == null) {
+                throw NavigationException("Must call start() and end() before calling getElapsedTime()")
+            }
+            val elapsedTimeInNanoseconds = end!! - start!!
+            val elapsedTimeInSeconds = elapsedTimeInNanoseconds / ELAPSED_TIME_DENOMINATOR
+            return (elapsedTimeInSeconds * PRECISION).roundToLong() / PRECISION
+        }
+
+    fun start() {
+        start = System.nanoTime()
```

Comment:
make separate `Time` interface for tests like 

```
interface Time {
  fun nanoTime(): Long
  fun mills(): Long
  object SystemImpl : Time {
     /// IMPL
  }
} 
```
`internal class ElapsedTime (private val time: Time = Time.SystemImpl) { ...}`

Reply:
I think it is a bit overengineering...